src/file_pipe_log/pipe_builder.rs

```diff
@@ -227,7 +227,10 @@ impl<F: FileSystem> DualPipesBuilder<F> {
                     let is_last_file = index == chunk_count - 1 && i == file_count - 1;
```

Comment:
You can extend the file list to include a version field. Try use [par_chunks_mut](https://docs.rs/rayon/1.0.3/rayon/slice/trait.ParallelSliceMut.html#method.par_chunks_mut), decode the version here and update the file list.

Reply:
Yes! It can't be agreed any more. Thanks, i'll take a try by introducing `par_chunks_mut` to modify the ugly implementation.
Firstly, when i tried to extend it, i found an embarrassing fact that the `file_list` was built by the `scan`, whose main purpose was to build the list of log files without loading `file header` in advance. Then, the validation of `header` was done during the processing of `recover`, which just got a `read only` `file_list`. That is, i could not update the `version` of each file here. So, the final is implemented as the below shown.